extension/js/common/api/key-server/attester.ts

```diff
@@ -25,7 +25,40 @@ export class Attester extends Api {
       console.info(`Skipping attester lookup of ${email} because attester search on this domain is disabled.`);
       return { pubkeys: [] };
     }
-    return await this.doLookup(email);
+    // first get from recipient-specific LDAP server, if any, relayed through flowcrypt.com
+    const customerLdapRes = await this.doLookupLdap(email);
```

Comment:
The issue description suggested to start all 3 requests simultaneously to reduce response time.
You can, of course, not wait for all of them to complete,
popping and awaiting each promise in a way like:
```
const requests = [
// in reverse order for easier popping
  this.doLookupLdap(email, 'keyserver.pgp.com'), // get from keyserver.pgp.com, relayed through flowcrypt.com,
  this.doLookup(email),  // get from flowcrypt.com public keyserver database
  this.doLookupLdap(email),  // get from recipient-specific LDAP server, if any, relayed through flowcrypt.com
];
while (requests.length) {
 const result = await requests.pop();
 if (result.pubkeys.length) {
     // but you need to set `.catch` handlers for the remaining promises to avoid possible `unhandled rejection`
    return result;
 }
}

Reply:
Perhaps, we can add something like this to the `common.ts` utils

Reply:
Problem is with this requirement.
We should first return customer-specific LDAP result (if there are any) and then flowcrypt server pub key results and then keyserver.pgp.com result

```
Then please choose and return only one set of results, with the following priority, starting from highest:

if at least one public key was returned from customer-specific LDAP, return an array of these public keys and nothing else
else, if flowcrypt.com public keyserver returned any public key, return only array of these
else, return array of keys returned from keyserver.pgp.com (or empty array)

```

Reply:
> but you need to set `.catch` handlers for the remaining promises to avoid possible `unhandled rejection`

in `finally` block, apparently

Reply:
Also with above way, we have to call 3 apis at once when we can call 1-2 apis.

Reply:
or set them up from the beginning, but bearing in mind, that they can be ignored in certain circumstances

Reply:
Problem with my implementation is user have to wait all 3 requests if there are no customer specific LDAP result, flowcrypt pub key server results.
But all requests only take less than 1 sec and this doesn't seem to be a problem I think

Reply:
They take less than 1 sec to localhost?

Reply:
> Problem is with this requirement. We should first return customer-specific LDAP result (if there are any) and then flowcrypt server pub key results and then keyserver.pgp.com result

I can't see what the problem is: `await Promise.all` returns an array of results in the same order as the order of promises in its argument array

Reply:
> Problem with my implementation is user have to wait all 3 requests if there are no customer specific LDAP result, flowcrypt pub key server results. But all requests only take less than 1 sec and this doesn't seem to be a problem I think

You can start them all, and then wait one-by-one by popping and `await`ing them individually (as I showed above) until the rest can be skipped

Reply:
No. To live api.

> They take less than 1 sec to localhost?

By the way, Sorry i misunderstood what you mean.
You are right. Will implement this way.

Reply:
> Also with above way [as suggested by Roman], we have to call 3 apis at once when we can call 1-2 apis.

I'm ok to call 3 APIs at once even if their results are not always used, as long as that makes the method sometimes finish faster. Don't want to let the user wait.